racket/src/bc/src/vector.c

```diff
@@ -1291,6 +1303,49 @@ static Scheme_Object *vector_star_append(int argc, Scheme_Object *argv[])
   return do_vector_append("vector*-append", 0, argc, argv);
 }
 
+static Scheme_Object *do_vector_extend(const char *who, int chaperone_ok, int argc, Scheme_Object *argv[])
+{
+  Scheme_Object *v, *fill, *new_vec, *argv2[2];
+  intptr_t new_len;
+
+  v = argv[0];
+  if (chaperone_ok && SCHEME_NP_CHAPERONEP(v))
+    v = SCHEME_CHAPERONE_VAL(v);
+  if (!SCHEME_VECTORP(v))
+    scheme_wrong_contract(who, "vector?", 0, argc, argv);
+  new_len = scheme_extract_index(who, 1, argc, argv, -1, 0);
+
+  if ((new_len == -1) 
+      /* also watch for overflow: */
+      || (REV_VECTOR_BYTES(VECTOR_BYTES(new_len)) != new_len)) {
+    scheme_raise_out_of_memory(who, "creating vector of length %s",
+			       scheme_make_provided_string(argv[1], 1, NULL));
+  }
+
+  if (argc == 3)
+    fill = argv[2];
+  else
+    fill = scheme_make_integer(0);
+
+  new_vec = scheme_make_vector(new_len - SCHEME_VEC_SIZE(v), fill);
```

Comment:
Is a check that `new_len > = SCHEME_VEC_SIZE(v)` needed, or is it somehow covered above?